src/main/java/jenkins/plugins/slack/SlackNotifier.java

```diff
@@ -389,46 +410,276 @@ public SlackNotifier(final String baseUrl, final String teamDomain, final String
         );
     }
 
+    @Deprecated
+    @CheckReturnValue
     public SlackNotifier(final String baseUrl, final String teamDomain, final String authToken, final boolean botUser, final String room, final String tokenCredentialId,
                          final String sendAs, final boolean startNotification, final boolean notifyAborted, final boolean notifyFailure,
                          final boolean notifyNotBuilt, final boolean notifySuccess, final boolean notifyUnstable, final boolean notifyRegression, final boolean notifyBackToNormal,
                          final boolean notifyRepeatedFailure, final boolean includeTestSummary, final boolean includeFailedTests, MatrixTriggerMode matrixTriggerMode,
                          CommitInfoChoice commitInfoChoice, boolean includeCustomMessage, String customMessage, String customMessageSuccess,
                          String customMessageAborted, String customMessageNotBuilt, String customMessageUnstable, String customMessageFailure) {
-        super();
-        this.baseUrl = baseUrl;
+
+        new SlackNotifier.SlackNotifierBuilder()
```

Comment:
this is the spotbugs issue, you're building a new object but not assigning it anywhere

Reply:
you should be calling this constructor instead with your builder:
```
private SlackNotifier(SlackNotifierBuilder slackNotifierBuilder) {
```